src/client/option.go

```diff
@@ -2,20 +2,25 @@ package client
 
 import "github.com/pachyderm/pachyderm/v2/src/pfs"
 
+type putFileConfig struct {
```

Comment:
Maybe we should just expose add file and have that be the append path. That would let us get rid of this, and put would always be delete followed by add.

Reply:
I think the client should mirror the CLI.  If we have put file in the CLI then we should have it in here too.

Reply:
I agree with that. I didn't mean get rid of put file, just this extra config layer that is used to handle the append case. If we exposed add, delete, and put then we can get rid of the append option for put.

Reply:
The CLI has append as an option on put though.

Reply:
Yeah, we can change that.

Reply:
We discussed this offline. We are going to circle back to this later.

Reply:
We spoke offline about this.  CLI and client facade changes are out of scope of this P.R.